### PR TITLE
[FEA] Add file size counter to cuIO benchmarks

### PR DESCRIPTION
**Pull request by vuule**
Most cuIO benchmarks used dataframes of fixed size as input. After writing to a file in the given format, its size can vary greatly depending on the encoding and compression.
This PR adds a counter to output the file size, as it can be often corelated with the performance of readers/writers.

### CODE REVIEW

**Comment by vuule** [comment]
Sample output:
```
OrcRead/integral_file_input/30/0/1/1/0/manual_time               97.2 ms         63.0 ms            7 bytes_per_second=5.1425G/s file_size=389.961M peak_memory_usage=1096.41M
OrcRead/integral_file_input/30/1000/1/1/0/manual_time             112 ms         82.1 ms            6 bytes_per_second=4.47958G/s file_size=331.69M peak_memory_usage=1.14553G
OrcRead/integral_file_input/30/0/32/1/0/manual_time              69.9 ms         67.2 ms           10 bytes_per_second=7.15121G/s file_size=22.2183M peak_memory_usage=683.382M
OrcRead/integral_file_input/30/1000/32/1/0/manual_time           69.6 ms         67.2 ms           10 bytes_per_second=7.18037G/s file_size=20.5541M peak_memory_usage=683.241M
OrcRead/integral_file_input/30/0/1/0/0/manual_time               81.7 ms         47.1 ms            8 bytes_per_second=6.12364G/s file_size=396.36M peak_memory_usage=951.602M
OrcRead/integral_file_input/30/1000/1/0/0/manual_time            78.0 ms         43.5 ms            9 bytes_per_second=6.40907G/s file_size=396.308M peak_memory_usage=951.549M
OrcRead/integral_file_input/30/0/32/0/0/manual_time              65.1 ms         62.0 ms           11 bytes_per_second=7.6859G/s file_size=24.9541M peak_memory_usage=580.196M
OrcRead/integral_file_input/30/1000/32/0/0/manual_time           64.5 ms         61.4 ms           11 bytes_per_second=7.75274G/s file_size=24.916M peak_memory_usage=580.158M
OrcRead/integral_buffer_input/30/0/1/1/1/manual_time              105 ms          105 ms            7 bytes_per_second=4.76301G/s file_size=389.961M peak_memory_usage=1096.41M
OrcRead/integral_buffer_input/30/1000/1/1/1/manual_time           118 ms          118 ms            6 bytes_per_second=4.24934G/s file_size=331.69M peak_memory_usage=1.14553G
OrcRead/integral_buffer_input/30/0/32/1/1/manual_time            68.8 ms         68.8 ms           10 bytes_per_second=7.27146G/s file_size=22.2183M peak_memory_usage=683.382M
OrcRead/integral_buffer_input/30/1000/32/1/1/manual_time         68.6 ms         68.7 ms           10 bytes_per_second=7.28465G/s file_size=20.5542M peak_memory_usage=683.241M
OrcRead/integral_buffer_input/30/0/1/0/1/manual_time             88.9 ms         88.9 ms            7 bytes_per_second=5.62704G/s file_size=396.36M peak_memory_usage=951.602M
OrcRead/integral_buffer_input/30/1000/1/0/1/manual_time          87.2 ms         87.2 ms            8 bytes_per_second=5.73492G/s file_size=396.308M peak_memory_usage=951.549M
OrcRead/integral_buffer_input/30/0/32/0/1/manual_time            63.9 ms         63.9 ms           11 bytes_per_second=7.8284G/s file_size=24.9541M peak_memory_usage=580.196M
OrcRead/integral_buffer_input/30/1000/32/0/1/manual_time         64.1 ms         64.1 ms           11 bytes_per_second=7.80321G/s file_size=24.916M peak_memory_usage=580.158M
```

File: cpp/benchmarks/io/csv/csv_reader.cpp [hunk 1]
```diff
@@ -132,6 +130,7 @@ void BM_csv_read_varying_options(benchmark::State& state)
   auto const data_processed = data_size * cols_to_read.size() / view.num_columns();
   state.SetBytesProcessed(data_processed * state.iterations());
   state.counters["peak_memory_usage"] = mem_stats_logger.peak_memory_usage();
+  state.counters["file_size"]         = source_sink.size();
```
Comment (vuule) [inline cpp/benchmarks/io/csv/csv_reader.cpp:133]: or "encoded_size" maybe?

**Comment by vuule** [comment]
CC @GregoryKimball who sort of asked for this feature

File: cpp/benchmarks/io/csv/csv_reader.cpp (hunk 1 above)
Comment (rgsl888prabhu) [reply cpp/benchmarks/io/csv/csv_reader.cpp:133]: "encoded_file_size" ?

**Review by rgsl888prabhu — COMMENTED** [review]
Rest looks good

**Review by rgsl888prabhu — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**